src/Common/Site_Health/Info_Section_Abstract.php

```diff
@@ -0,0 +1,232 @@
+			'description' => $this->filter_param( 'description', $this->get_description() ),
+			'private'     => $this->filter_param( 'private', $this->is_private() ),
+			'show_count'  => $this->filter_param( 'show_count', $this->get_show_count() ),
+			'fields'      => $this->filter_param( 'fields', $fields ),
+		];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_label(): string {
+		return $this->label;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_description(): string {
+		return $this->description;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_fields(): array {
+		return $this->fields;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_show_count(): bool {
+		return $this->show_count;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function is_private(): bool {
+		return $this->is_private;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function has_field( $field ): bool {
+		// Assume field is a key first.
+		$id = $field;
+
+		// When it's a field use its ID.
+		if ( $field instanceof Info_Field_Abstract ) {
+			$id = $field->get_id();
+		}
+
+		// Keys can only be strings.
+		if ( ! is_string( $id ) ) {
+			return false;
+		}
+
+		$fields = $this->get_fields();
+
+		return isset( $fields[ $id ] );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function get_field( string $id ): ?Info_Field_Abstract {
+		if ( ! $this->has_field( $id ) ) {
+			return null;
+		}
+
+		return $this->get_fields()[ $id ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function add_field( Info_Field_Abstract $field, bool $overwrite = false ): bool {
+		// Allow the adding of a field to overwrite existing fields.
+		if ( ! $overwrite && $this->has_field( $field ) ) {
+			return false;
+		}
+
+		$this->fields[ $field->get_id() ] = $field;
+
+		$this->sort_fields();
+
+		return true;
+	}
+
+	/**
+	 * Internal method to the Section class, to allow filtering of specific values.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $param
+	 * @param mixed  $value
+	 *
+	 * @return mixed
+	 */
+	protected function filter_param( string $param, $value = null ) {
+		$section_slug = static::get_slug();
+		/**
+		 * Filters the get of a particular param for all sections.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed                 $value   Value of the field.
+		 * @param Info_Section_Abstract $section Current Section.
+		 */
+		$value = apply_filters( "tec_debug_info_section_get_{$param}", $value, $this );
+
+		/**
+		 * Filters the get of a particular param for a specific section.
+		 *
+		 * @since TBD
+		 *
+		 * @param mixed                 $value   Value of the field.
+		 * @param Info_Section_Abstract $section Current Section.
+		 */
+		return apply_filters( "tec_debug_info_section_{$section_slug}_get_{$param}", $value, $this );
+	}
+
+	/**
+	 * Sorts the fields stored on this section, will retain keys.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	protected function sort_fields(): void {
```

Comment:
Be more explicit - sort by what? Where?
```suggestion
	/**
	 * Sorts the fields stored on this section in-place by priority. Will retain keys.
	 *
	 * @since TBD
	 *
	 * @return void
	 */
	protected function sort_fields_by_priority(): void {
```